lua/lspconfig/server_configurations/solc.lua

```diff
@@ -0,0 +1,27 @@
+local util = require 'lspconfig.util'
+
+local bin_name = 'solc-static-linux'
+if vim.fn.has 'mac' then
+  bin_name = 'solc-macos'
+end
+if vim.fn.has 'win32' == 1 then
+  bin_name = 'solc-windows'
+end
+
+return {
+  default_config = {
+    cmd = { bin_name, '--lsp' },
+    filetypes = { 'solidity' },
+    root_dir = util.root_pattern('.git'),
```

Comment:
No idea if this is a proper root_dir pattern. Copied it from `solidity_ls` and just removed `package.json`.